figo/figo.py

```diff
@@ -129,23 +129,31 @@ class FigoException(Exception):
     They consist of a code-like `error` and a human readable `error_description`.
     """
 
-    def __init__(self, error, error_description, code):
+    def __init__(self, error, error_description, code = None):
         """Create a Exception with a error code and error description."""
         super(FigoException, self).__init__()
 
+        # XXX(dennis.lutter): not needed internally but left here for backwards compatibility
         self.error = error
         self.error_description = error_description
         self.code = code
 
     def __str__(self):
         """String representation of the FigoException."""
-        return "FigoException: %s(%s)" % (repr(self.error_description), repr(self.error))
+        return "FigoException: {}({})" .format(self.error_description, self.error)
 
     @classmethod
     def from_dict(cls, dictionary):
         """Helper function creating an exception instance from the dictionary returned
         by the server."""
-        return cls(dictionary['error']['message'], dictionary['error']['description'], dictionary['error']['code'])
+        if 'code' in dictionary['error']:
```

Comment:
True that! 👍  
### Pitfall of `.get(some_attr)`
However, the `.get(some_attr)` doesn't throw a KeyError, but simply evaluates to `None`. As `FigoException`s `__init__` takes `error` and `error_description` as mandatory arguments, this will cause the code to break in case one or both arguments are missing in a malformed HTTP response (I've seen various unexpected behaviour while working with the figo connect API) .

### Solution
To make sure this doesn't happen, we should use `.get(some_attr, alternative)` especially for `error` and `error_description` to pass in an empty string and maybe also for the error `code` to pass in an unspecified error.

So what do you think? I'll be glad to fix this entirely and include it in the PR